python/mlc_chat/compiler/quantization/group_quantization.py

```diff
@@ -102,24 +149,44 @@ def _dequantize(
         return dequantized_weight
 
     def quantize_weight(self, weight: NDArray) -> List[NDArray]:
-        """Quantize weight with group quantization"""
+        """
+        Quantize weight with group quantization
+
+        Parameters
+        ----------
+        weight : NDArray
+            The original weight.
+
+        Returns
+        ------
+        ret: List[NDArray]
+            The list of group quantized weights.
+        """
         assert weight.dtype == self.model_dtype
         assert len(weight.shape) == 2
-        bb = relax.BlockBuilder()
+        bb = relax.BlockBuilder()  # pylint: disable=invalid-name
         weight_var = relax.Var("weight", relax.TensorStructInfo(weight.shape, self.model_dtype))
         with bb.function(name="quantize", params=[weight_var]):
             with bb.dataflow():
-                lv = bb.emit_te(self._quantize, weight_var)
-                gv = bb.emit_output(lv)
+                lv = bb.emit_te(self._quantize, weight_var)  # pylint: disable=invalid-name
+                gv = bb.emit_output(lv)  # pylint: disable=invalid-name
             bb.emit_func_output(gv)
         mod = bb.get()
-        with Target("cuda"):
-            mod = dl.ApplyDefaultSchedule(  # pylint: disable=not-callable
-                dl.gpu.Reduction(), dl.gpu.GeneralReduction(), dl.gpu.Fallback()
-            )(mod)
-        ex = relax.build(mod, "cuda")
-        dev = device("cuda", 0)
-        vm = relax.VirtualMachine(ex, dev)
+        dev = weight.device
+        device_type = dev.MASK2STR[dev.device_type]
+        if device_type in ["cuda", "rocm", "metal", "vulkan"]:
+            target = Target.from_device(dev)
+            with target:
+                mod = dl.ApplyDefaultSchedule(  # pylint: disable=not-callable
+                    dl.gpu.Reduction(), dl.gpu.GeneralReduction(), dl.gpu.Fallback()
```

Comment:
why only reduction and general reduction? i suppose matmul and gemv are [missing](https://github.com/mlc-ai/mlc-llm/blob/9ad88baf5161bfb7d6d8e4e6f4e36bc589ada9cb/python/mlc_chat/compiler/compiler_pass/pipeline.py#L57)?

Reply:
for group quantization, there is no matmul and gemv I think.